src/algorithms/synthesis/dd_synthesis.cpp

```diff
@@ -133,19 +133,39 @@ namespace syrec {
         }
     }
 
-    // This function stores all the controls of the `current` node concerning the root/src of the DD.
-    auto DDSynthesizer::controlRoot(dd::mEdge const& current, dd::Controls& ctrl, TruthTable::Cube const& ctrlCube) -> void {
-        const auto cubeSize = ctrlCube.size();
-        for (auto i = 0U; i < cubeSize; ++i) {
-            if (ctrlCube[i].has_value()) {
-                const auto idx = static_cast<dd::Qubit>((cubeSize - i) + static_cast<std::size_t>(current.p->v));
-                if (*ctrlCube[i]) {
-                    ctrl.emplace(dd::Control{idx, dd::Control::Type::pos});
-                } else {
-                    ctrl.emplace(dd::Control{idx, dd::Control::Type::neg});
-                }
+    // This function stores all the ESOP minimized version of all the controls of the `current` node not concerning the root/src of the DD.
+    auto DDSynthesizer::controlNonRootEsop(dd::mEdge const& current, dd::Controls& ctrl, minbool::MinTerm const& ctrlCube, std::size_t const& n2) -> void {
+        for (int j = static_cast<int>(n2) - 1; j >= 0; --j) {
+            if (ctrlCube[j] == minbool::MinTerm::Value::One) {
+                ctrl.emplace(dd::Control{static_cast<dd::Qubit>(static_cast<std::size_t>(current.p->v) - (static_cast<int>(n2) - 1 - j + 1)), dd::Control::Type::pos});
+            } else if (ctrlCube[j] == minbool::MinTerm::Value::Zero) {
+                ctrl.emplace(dd::Control{static_cast<dd::Qubit>(static_cast<std::size_t>(current.p->v) - (static_cast<int>(n2) - 1 - j + 1)), dd::Control::Type::neg});
+            }
+        }
+    }
+
+    // This function stores all the (ESOP minimized version) controls of the `current` node concerning the root/src of the DD.
+    auto DDSynthesizer::controlRootEsop(dd::mEdge const& current, dd::Controls& ctrl, minbool::MinTerm const& ctrlCube, std::size_t const& n2) -> void {
+        for (int j = static_cast<int>(n2) - 1; j >= 0; --j) {
+            if (ctrlCube[j] == minbool::MinTerm::Value::One) {
+                ctrl.emplace(dd::Control{static_cast<dd::Qubit>((n2 + static_cast<std::size_t>(current.p->v)) - (static_cast<int>(n2) - 1 - j)), dd::Control::Type::pos});
+            } else if (ctrlCube[j] == minbool::MinTerm::Value::Zero) {
+                ctrl.emplace(dd::Control{static_cast<dd::Qubit>((n2 + static_cast<std::size_t>(current.p->v)) - (static_cast<int>(n2) - 1 - j)), dd::Control::Type::neg});
+            }
+        }
+    }
+
+    // This function returns the ESOP minimized form of a signature.
+    auto DDSynthesizer::esopOptimization(TruthTable::Cube::Vector const& sigVec) -> std::vector<minbool::MinTerm> {
+        std::vector<uint64_t> on;
+        on.reserve(sigVec.size());
+
+        for (auto const& sigVecToInt: sigVec) {
+            if (!sigVecToInt.empty()) {
+                on.emplace_back(sigVecToInt.toInteger());
             }
         }
+        return minbool::minimizeBoolean(on, sigVec[0].size());
```

Comment:
Why does this function not handle don't care conditions? In anticipation of the future this would make a lot of sense and is definitely supported by the underlying library.

Reply:
Currently, the don't care conditions are not required. Also, for the future based on the technique that I employ in the synthesis technique, I don't think we will require this as well. I have removed some statements and functions that are concerning the `dcValues` since it is not required. 

if dc conditions are inevitably required, I will add those statements and functions back in the future commits.

Reply:
Sounds good to me. The removal of dc handling should make some of the functions in the minimization a lot easier.